### ISSUE TITLE
with-firebase-hosting-and-typescript not working

### ISSUE THREAD

**Issue by Industrial** [zeit/next.js#9291]
# Bug report

## Describe the bug

The deployment fails

## To Reproduce

1. `yarn create next-app --example with-firebase-hosting-and-typescript with-firebase-hosting-and-typescript-app`
2. `yarn deploy`

```
Error: > Couldn't find a `pages` directory. Please create one under the project root
    at findPagesDir (/home/tom/Code/Code9/Tests/with-firebase-hosting-and-typescript-app/dist/functions/node_modules/next/dist/lib/find-pages-dir.js:3:170)
    at new DevServer (/home/tom/Code/Code9/Tests/with-firebase-hosting-and-typescript-app/dist/functions/node_modules/next/dist/server/next-dev-server.js:1:3301)
    at Object.createServer [as default] (/home/tom/Code/Code9/Tests/with-firebase-hosting-and-typescript-app/dist/functions/node_modules/next/dist/server/next.js:2:105)
    at Object.<anonymous> (/home/tom/Code/Code9/Tests/with-firebase-hosting-and-typescript-app/dist/functions/index.js:16:27)
    at Module._compile (internal/modules/cjs/loader.js:776:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:787:10)
    at Module.load (internal/modules/cjs/loader.js:653:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:593:12)
    at Function.Module._load (internal/modules/cjs/loader.js:585:3)
    at Module.require (internal/modules/cjs/loader.js:690:17)
```

**Comment by Industrial** [comment]
This also happens with the `with-firebase-hosting` version, which might reduce the reproduction size.

Maybe it has to do with the version of Next.js (9.x)?

**Comment by Industrial** [comment]
The deployment succeeded when I changed the value of `dev` to `false`

**Comment by Industrial** [comment]
1. Change `.firebaserc` to use your Firebase Project ID.
2. Change `package.json` to use node v10 instead of v8. Firebase supports v10.
3. Run `yarn`.
4. Get some tslint errors about code style. Fix them. I did this in https://github.com/zeit/next.js/pull/9318
5. Now throws `Error: > Couldn't find a `pages` directory. Please create one under the project root`

**Comment by Industrial** [comment]
6. in `src/functions/index.ts` change `next({ dev, conf: { distDir: 'next' } })` to `next({ dev: false, conf: { distDir: 'next' } })`.

Now the process continues to the point where it says:
```
i  functions: preparing dist/functions directory for uploading...
```

and then it throws

```
Error: Could not find a valid build in the '/some/dir/next.js/examples/with-firebase-hosting-and-typescript/next' directory! Try building your app with 'next build' before starting the server.
```

**Comment by Industrial** [comment]
@timneutkens Still having the deployment issue

**Comment by timneutkens** [comment]
@Industrial okay? I've already added labels right?

**Comment by Industrial** [comment]
My problem has ✨magically✨ disappeared, and I have a successful deploy.

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
